myuw/dao/student.py

```diff
@@ -0,0 +1,143 @@
+"""
+This module will provide a method to retrieve all user majors, class standings
+and minors along with current and rollup variables to facilitate
+determinations about user context in the case that a user is not
+enrolled for the current quarter
+"""
+from myuw.dao.enrollment import enrollment_search
+from uw_sws.term import get_current_term
+
+
+def get_student_status(request):
+    """
+    This will return a dictionary with the student status over time alongside
+    various rollup variables
+    :param request:
+    :return: dict
+    """
+    enrollments = enrollment_search(request)
+
+    return _process_fields(enrollments, ['majors', 'minors', 'class_level'])
+
+
+def get_majors(request):
+    """
+    Returns a dictionary with a dict of majors by terms in the 'majors'
+    attribute, and a 'current' and 'rollup' major field
+    :param request
+    :return: dict
+    """
+    enrollments = enrollment_search(request)
+
+    return _get_majors(enrollments)
+
+
+def get_rollup_and_future_majors(majors):
+    return _get_rollup_and_future(majors['majors'], majors['rollup'])
+
+
+def _get_majors(enrollments):
+    return _process_fields(enrollments, ["majors"])['majors']
+
+
+def get_minors(request):
+    """
+    Returns a dictionary with a dict of majors by terms in the 'majors'
+    attribute, and a 'current' and 'rollup' major field
+    :param request
+    :return: dict
+    """
+    enrollments = enrollment_search(request)
+
+    return _get_minors(enrollments)
+
+
+def get_rollup_and_future_minors(minors):
+    return _get_rollup_and_future(minors['minors'], minors['rollup'])
+
+
+def _get_minors(enrollments):
+    return _process_fields(enrollments, ["minors"])['minors']
+
+
+def get_class_standings(request):
```

Comment:
Could you tell me where this function is used?

Reply:
I don't have a current use for get_class_standing but I had thought that there may be a future feature wherein we run into a similar issue where class standing is not available to a module due to a lack of current enrollment data